app/controllers/submissions/new.js

```diff
@@ -93,17 +93,13 @@ export default Controller.extend({
       sub.set('source', 'pass');
       sub.set('removeNIHDeposit', false);
       sub.set('aggregatedDepositStatus', 'not-started');
-      sub.set(
-        'repositories',
-        sub.get('repositories').filter(repo =>
-          // TODO the specific URL checks should be removed after Repository data is updated to
-          // include 'integrationType' property
-          (
-            repo.get('integrationType') !== 'web-link' &&
-            repo.get('url') !== 'https://eric.ed.gov/' &&
-            repo.get('url') !== 'https://dec.usaid.gov/'
-          ))
-      );
+
+      if (!this.get('hasProxy')) {
+        sub.set(
+          'repositories',
+          sub.get('repositories').filter(repo => (repo.get('integrationType') !== 'web-link'))
+        );
+      }
```

Comment:
I think this block of filtering out weblink repo should happen in the block of line 47. because what we want is for web-link repo to be remove JUST before a submission is submitted for the last time.

Reply:
@htpvu 

The is what already happens. The code you quote above does exactly that.

Reply:
I suppose hasProxy could be buggy? But that probably means there are other issues. I think I tested by doing just a straight submission, but should try again.